libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -258,6 +258,9 @@ const AP_Param::GroupInfo AP_GPS::var_info[] = {
 // constructor
 AP_GPS::AP_GPS()
 {
+    static_assert((sizeof(_initialisation_blob) * 9) < (4800 * GPS_BAUD_TIME_MS * 1e-3),
```

Comment:
I don't think there would be a problem with that as I would expect the UART driver to still use 8N1 and convert it to a byte with whatever bits it would have in that platform. The biggest problem we would encounter is we extensively use uint8 across our codebase and that won't exist in platforms that have a CHAR_BIT bigger than 8.

Anyway, my suggestion wasn't obviously because of portability, but to try and make it more readable without adding a new macro. Let me know when you've made the change and I'll merge.